### PR TITLE
[ADF-5158] ECM Version Directive

### PR DESCRIPTION
**Pull request by davidcanonieto**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
https://issues.alfresco.com/jira/browse/ADF-5158

**What is the new behaviour?**
This new directive allows component to hide when the app is not running the ACS version required for them to work properly. 

```html
<button *adf-ecm-version="'6.0.0'">
    My Action
</button>
```
This means this button will be present in the DOM tree from 6.0.0 onwards. Otherwise it will be removed to avoid the component breaking the app.

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/core/services/version-compatibility.service.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        private alfrescoApiService: AlfrescoApiService,
+        private discoveryApiService: DiscoveryApiService
+    ) {
+        this.alfrescoApiService.alfrescoApiInitialized
+            .pipe(filter(status => status))
+            .subscribe(this.initCompatibilityVersion.bind(this));
+    }
+
+    private initCompatibilityVersion() {
+        this.discoveryApiService.getEcmProductInfo().toPromise().then(
```
Comment (DenysVuika) [inline lib/core/services/version-compatibility.service.ts:40]: don't use `.toPromise` because it is deprecated and will be soon removed

**Comment by DenysVuika** [comment]
I don't think "ECM" (enterprise content management) is the right term here, maybe "ACS" is better?

**Comment by eromano** [comment]
I don't think you need to change the e2e they are working on develop

**Pull request merged**